stdsimd/arch/detect/arch/x86.rs

```diff
@@ -16,8 +16,22 @@
 //! in a global `AtomicUsize` variable. The query is performed by just checking
 //! whether the feature bit in this global variable is set or cleared.
 
+/// A macro to test at *runtime* whether a CPU feature is available on
+/// x86/x86-64 platforms.
+///
+/// This macro is provided in the standard library and will detect at runtime
+/// whether the specified CPU feature is detected. This does *not* resolve at
+/// compile time unless the specified feature is already enabled for the entire
+/// crate. Runtime detection currently relies mostly on the `cpuid` instruction.
+///
+/// This macro only takes one argument which is a string literal of the feature
+/// being tested for. The feature names supported are the lowercase versions of
+/// the ones defined by Intel in [their documentation][docs].
+///
```

Comment:
Do we want to list the whitelisted features here?

Reply:
Maybe also mentions that these names are the same as those used by `#[target_feature]`, and maybe it might be also worth mentioning twice that `is_x86_feature_detected("avx,avx2")` is not supported (even though it works for target feature.

Reply:
Excellent ideas!